### PR TITLE
Adding support for an end-date argument.

### PR DESCRIPTION
**Pull request by burkematthew**
Addresses #278 and #208.  This is my first contribution to a project like this, so feedback is certainly welcome.

### CODE REVIEW

**Comment by burkematthew** [comment]
Addresses #278 and #208.  This is my first contribution to a project like this, so feedback is certainly welcome.

File: mintapi/api.py [hunk 1]
```diff
@@ -789,29 +799,22 @@ def get_transactions_json(self, include_investment=False,
             txns = data['set'][0].get('data', [])
             if not txns:
                 break
-            if start_date:
```
Comment (mrooney) [inline mintapi/api.py:792]: Is the idea here that we can use startDate and endDate directly in the calls above, and don't need to filter ourselves anymore?

File: mintapi/api.py [hunk 2]
```diff
@@ -760,6 +761,13 @@ def get_transactions_json(self, include_investment=False,
             start_date = datetime.strptime(start_date, '%m/%d/%y')
         except (TypeError, ValueError):
             start_date = None
+
+        # Converts the end date into datetime format - must be mm/dd/yy
+        try:
+            end_date = datetime.strptime(end_date, '%m/%d/%y')
+        except (TypeError, ValueError):
+            end_date = None
+
```
Comment (mrooney) [inline mintapi/api.py:770]: Makes sense! My only thought here is that this is basically copy and pasted from above. Can this be abstracted into a little helper method and then you can simply call it twice with different dates?

**Review by mrooney — CHANGES_REQUESTED** [review]
Thanks for your awesome contribution! Looks great overall, just a minor question / feedback or two.

File: mintapi/api.py (hunk 1 above)
Comment (burkematthew) [reply mintapi/api.py:792]: That's correct.  I was running into some friction with the fact that we were using the filtering to also break out of the loop.  So I couldn't exactly do the same thing that we did with start_date.  I explored a little more of the methods and saw that there was another instance of using startDate and endDate.  I then went ahead and tested out the behavior myself and got the desired results.

File: mintapi/api.py (hunk 2 above)
Comment (burkematthew) [reply mintapi/api.py:770]: Yes, definitely can!  I will do that.

**Review by mrooney — APPROVED** [review]
Looks awesome, thanks!

**Review by mrooney — CHANGES_REQUESTED** [review]
Actually, I realized I had to approve the workflow checks for you as a first-time contributor, and it did hit very minor flake8 line spacings to fix: https://github.com/mintapi/mintapi/pull/301/checks?check_run_id=3810069327.

You should be able to run that locally and add the extra spaces, or if not, just add them on the lines mentioned :) Thanks again and looking forward to merging and releasing this!

**Pull request merged**